util/dvsim/doc/testplanner.md

```diff
@@ -0,0 +1,285 @@
+---
+title: "testplanner tool"
+---
+
+testplanner is a Python based tool for parsing testplans written in Hjson
+format into a data structure that can be used for:
+* Expanding the testplan inline within the DV document as a table
+* Annotating the simulation results with testplan entries for a document driven
+  DV execution
+
+Please see [DV methodology]({{< relref "doc/ug/dv_methodology/index.md#documentation" >}})
+for more details on the rationale and motivation for writing and maintaining testplans
+in a machine-parsable format (`Hjson`).
+This document will focus on the anatomy of a Hjson testplan, list of features supported
+and some of the ways of using the tool.
+
+## Hjson testplan
+
+A testplan consists of a list of testpoints (planned tests) and a list of
+planned coverage items (covergroups).
+
+### Testpoints
+
+A testpoint is an entry in the testplan representing a planned test. A planned
+test maps 1:1 to a uniquely testable feature of the design. Minimally, the
```

Comment:
Hmm, I am afraid the comments above are missing the gist of this PR. The counter-examples mentioned are "tests" (stress_all, smoke). In my opinion the distinction between testpoints and tests is valid. I tend to think of what is referred to here as "testpoint" as a "design feature"; for example, AES should encrypt per the spec, and AES should decrypt per the spec for the block sizes supported. These testpoints can be verified by any number of tests DV engineers decide to write.

Reply:
@matutem: I understand the distinction that you're making, but I think the things listed in this file under the name "testpoint" *are* stress_all, smoke etc. Which is why I'm saying that this might be mis-named.

Reply:
Hmm? Each testpoint can include a list of tests, which can be any of the tests written, and that is just fine: we need to attach actual tests somewhere. I also gather any given test can be attached to multiple testpoints, which also makes sense. But I doubt we should create a testpoint for a generic test like stress_all. Perhaps Sri can clarify this further?

Reply:
Thanks, this is clearer, I think.